### DISCUSSION TITLE
Start error when run strongswan on Ubuntu firefly

### DISCUSSION THREAD

**Discussion by goodygamee** [strongswan/strongswan#564] [General]
- OS: [Firefly based on Ubuntu]
 - strongSwan version(s): [e.g. 5.9.0]

When running strongswan in debug mode, an error occured:  "received netlink error: operation not supported (95)", detailed below:

root@firefly:~# /usr/local/libexec/ipsec/starter --nofork
Starting strongSwan 5.9.0 IPsec [starter]...
00[DMN] Starting IKE charon daemon (strongSwan 5.9.0, Linux 4.4.179, aarch64)
00[LIB] created TUN device: ipsec0
create receive information sa thread
00[KNL] received netlink error: Operation not supported (95)
00[KNL] unable to create IPv4 routing table rule
00[KNL] received netlink error: Operation not supported (95)
00[KNL] unable to create IPv6 routing table rule
00[CFG] loading ca certificates from '/usr/local/etc/ipsec.d/cacerts'
00[LIB]   file coded in unknown format, discarded
00[LIB] building CRED_CERTIFICATE - X509 failed, tried 3 builders
00[CFG]   loading ca certificate from '/usr/local/etc/ipsec.d/cacerts/ca.cert.pem' failed
00[CFG] loading aa certificates from '/usr/local/etc/ipsec.d/aacerts'
00[CFG] loading ocsp signer certificates from '/usr/local/etc/ipsec.d/ocspcerts'
00[CFG] loading attribute certificates from '/usr/local/etc/ipsec.d/acerts'
00[CFG] loading crls from '/usr/local/etc/ipsec.d/crls'
00[CFG] loading secrets from '/usr/local/etc/ipsec.secrets'
00[LIB]   file coded in unknown format, discarded
00[LIB] building CRED_PRIVATE_KEY - RSA failed, tried 4 builders
00[CFG]   loading private key from '/usr/local/etc/ipsec.d/private/server.pem' failed

**Comment by tobiasbrunner** [comment]
> 00[KNL] received netlink error: Operation not supported (95)
> 00[KNL] unable to create IPv4 routing table rule

This requires the `CONFIG_IP_MULTIPLE_TABLES` kernel option (see [KernelModules](https://wiki.strongswan.org/projects/strongswan/wiki/KernelModules)).

There are other issues too (certificate/key loading). Are the files actually invalid or are plugins missing? Is the creation of `ipsec0` (i.e. loading the [_kernel-libipsec_ plugin](https://wiki.strongswan.org/projects/strongswan/wiki/kernel-libipsec)) intentional? The `create receive information sa thread` is not from an upstream component, so there might be modifications to the source code. Also, the ARM64 architecture might affect this (e.g. in regards to kernel options).

**Comment by goodygamee** [comment]
Thanks for your information. We'll try to fix this problem(ARM64 architecture server indeed) and share you new progress or bugs. Happy Monday!

----- 原始邮件 -----
发件人: "Tobias Brunner" ***@***.***>
收件人: "strongswan/strongswan" ***@***.***>
抄送: "goodygamee" ***@***.***>, "Author" ***@***.***>
发送时间: 星期四, 2021年 8 月 26日 下午 3:55:25
主题: Re: [strongswan/strongswan] Start error when run strongswan on Ubuntu firefly (#564)

> 00[KNL] received netlink error: Operation not supported (95)
> 00[KNL] unable to create IPv4 routing table rule

This requires the `CONFIG_IP_MULTIPLE_TABLES` kernel option (see [KernelModules](https://wiki.strongswan.org/projects/strongswan/wiki/KernelModules)).

There are other issues too (certificate/key loading). Are the files actually invalid or are plugins missing? Is the creation of `ipsec0` (i.e. loading the [_kernel-libipsec_ plugin](https://wiki.strongswan.org/projects/strongswan/wiki/kernel-libipsec)) intentional? The `create receive information sa thread` is not from an upstream component, so there might be modifications to the source code. Also, the ARM64 architecture might affect this (e.g. in regards to kernel options).


-- 
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub:
https://github.com/strongswan/strongswan/discussions/564#discussioncomment-1237763
